Test/Tests/RealtimeClientPresenceTests.swift

```diff
@@ -820,12 +812,13 @@ class RealtimeClientPresenceTests: XCTestCase {
                 done()
             }
         }
-
+        
         var presenceEvents = [ARTPresenceMessage]()
-
+        
         waitUntil(timeout: testTimeout) { done in
             let partialDone = AblyTests.splitDone(4, done: done)
             mainChannel.presence.subscribe { presence in
+                guard presence.clientId != mainClient.clientId else { return } // ignore events from themselves
```

Comment:
Why — surely we want to catch any incorrect presence events that are emitted for `mainClient.clientId`? Is this so that we don't call `partialDone()` again when we receive the fake synthesised presence `LEAVE` that you inject below? If so, would it not be better to simply unsubscribe from presence before calling `partialDone()` here?

Reply:
It's the second `enter` for the "mainClient" after its connection re-established.

Reply:
Hmm, OK. Can you add a comment, please?

Reply:
https://github.com/ably/ably-cocoa/pull/1714/commits/a00f89bfd054b160ac5ee7a7251b36ec042838b1